barretenberg/cpp/src/barretenberg/ultra_honk/ultra_verifier.cpp

```diff
@@ -79,11 +79,12 @@ template <typename Flavor> bool UltraVerifier_<Flavor>::verify_proof(const HonkP
     // Extract nested pairing points from the proof
     // TODO(https://github.com/AztecProtocol/barretenberg/issues/1094): Handle pairing points in keccak flavors.
     if constexpr (!std::is_same_v<Flavor, UltraKeccakFlavor> && !std::is_same_v<Flavor, UltraKeccakZKFlavor>) {
-        std::span<FF, PAIRING_POINTS_SIZE> pairing_points_limbs{
-            verification_key->public_inputs.data() +
-                verification_key->verification_key->pairing_inputs_public_input_key.start_idx,
-            PAIRING_POINTS_SIZE
-        };
+        const size_t limb_offset = verification_key->verification_key->pairing_inputs_public_input_key.start_idx;
+        BB_ASSERT_GTE(verification_key->public_inputs.size(),
```

Comment:
This catches an issue in starknet flavour - I didn't dig more, but good to have in